### ISSUE TITLE
Using `type_name` in .swiftlint.yml causes incorrect style errors

### ISSUE THREAD

**Issue by ArtisOracle** [realm/SwiftLint#526]
## Abstract
If I try to control how `type_name` rule (length) behaves in my project, any type that IS legal throws an error when run against the linter.

### Steps for reproduction
1. Open an Xcode project that has some types declared
2. Add swiftlint
3. Touch `.swiftlint.yml` file
4. Paste the following into the yml file:

```YML
disabled_rules: # rule identifiers to exclude from running
  - trailing_whitespace
  - closing_brace
  - variable_name
  - nesting
  
type_name:
    min_length: 2
    max_length: 40

variable_name:
	min_length: 500
	max_length: 1000

file_length:
    warning: 1000
    error: 1500
```

Then compile the project with swiftlint enabled

### Expected behavior
Any type names less than two characters and greater than forty characters are flagged as a warning.

### Actual behavior
Type names that do fit in the 2-40 range are marked as an error:

![screen shot 2016-02-11 at 5 09 24 pm](https://cloud.githubusercontent.com/assets/1455368/12992504/e04b3090-d0e2-11e5-96a2-d84fc4de7447.png)
![screen shot 2016-02-11 at 5 10 07 pm](https://cloud.githubusercontent.com/assets/1455368/12992503/e02f15b8-d0e2-11e5-90a6-bfa4d972f6c9.png)

### Workaround
Remove the rule from the .swiftlint.yml file and no errors are generated.

### Attempted resolutions:
- Tried changing the `type_name` rule to:

```YML
type_name:
    min_length: 2
    max_length:
		warning: 40
		error: 50
```

 This caused swiftlint to ignore the rule altogether.

### Version
    $ swiftlint version
    0.8.0

I am not sure if my yml format is invalid or if something else is going wrong. This was not occurring until I `brew upgrade`'d today to 0.8.0.

Any help would be greatly appreciated. And thank you for this tool and the help.

**Comment by scottrhoyt** [comment]
@jpsim do you think this is the same error as #522 ?

**Comment by jpsim** [comment]
Yes, this is the same as #522 which has been fixed on `master` and will be included in the next release.

Thanks for reporting @ArtisOracle!

**Comment by ArtisOracle** [comment]
Thank you!!
